### ISSUE TITLE
[Bug]: addon-a11y runs in docs mode and slows down large docs pages

### ISSUE THREAD

**Issue by kroeder** [storybookjs/storybook#30975]
### Describe the bug

We recently discovered in our internal Storybook that a large story in docs mode with a relatively complex component (many elements to check for axe.js) got significantly slower to a point where axe.js is a browser blocking task. Checking each story in story mode runs fast but in docs mode, it is just too much for axejs it seems. To confirm this, I disabled addon-a11y and the story was instantly rendered with no input lags.

We can't tell why addon-a11y suddenly is slow for this story (multiple seconds input lag until the page is fully loaded) but one solution would be to just not run addon-a11y in docs mode. Is there any reason it should run in docs mode? There are no controls to check for a11y violations and to me, it seems it would only make sense to have it turned on in story mode.

If you agree, I'll provide a small PR to fix this.

### Reproduction link

https://stackblitz.com/edit/github-q96uz5cy?file=src%2Fstories%2FButton.mdx

### Reproduction steps

1. Go to stackblitz example
2. Open dev console
3. Open button story in docs mode
4. See axe timing logs

### System

```bash
System:
    OS: macOS 15.3.2
    CPU: (8) arm64 Apple M1 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.13.1 - ~/.nvm/versions/node/v22.13.1/bin/node
    Yarn: 1.22.22 - /opt/homebrew/bin/yarn
    npm: 10.9.2 - ~/.nvm/versions/node/v22.13.1/bin/npm <----- active
    pnpm: 9.15.4 - ~/.nvm/versions/node/v22.13.1/bin/pnpm
  Browsers:
    Chrome: 134.0.6998.166
    Edge: 134.0.3124.85
    Safari: 18.3.1
  npmPackages:
    @storybook/addon-a11y: 8.6.11 => 8.6.11 
    @storybook/addon-essentials: 8.6.11 => 8.6.11 
    @storybook/addon-interactions: 8.6.11 => 8.6.11 
    @storybook/angular: 8.6.11 => 8.6.11 
    @storybook/blocks: ^8.6.11 => 8.6.11 
    @storybook/icons: ^1.4.0 => 1.4.0 
    @storybook/test: ^8.6.11 => 8.6.11 
    @storybook/theming: ^8.6.11 => 8.6.11 
    chromatic: ^11.12.6 => 11.12.6 
    eslint-plugin-storybook: ^0.10.1 => 0.10.1 
    storybook: ^8.4.2 => 8.6.11
```

**Comment by kroeder** [comment]
Since the change was no effort (besides having tons of issues with the Storybook repo setup...) feel free to either accept or close the PR

If you think running addon-a11y in docs mode make sense, I'll try to create a repo with a relatively complex component (let's hope AI can help me out..) and try to reproduce the slow down.

**Comment by smlimasu** [comment]
Can confirm the same issue for a few days. One table heavy page freezed completely. Seeing axe in the performance panel as the culprit.

I've found that instead of disabling the complete addon, another workaround was to set the globals.a11y.manual to true, so that the accessibility addon panel can still be run via click on the little button in the story view. 

Hint: We had the issues suddenly without upgrade being on 8.5.5 for a while, but persisted after upgrading to 8.6.10

**Comment by guicara** [comment]
We’re experiencing the same issue with large stories/components (like a MUI Table).

The browser almost freezes when rendering the story. Hover effects and interactions appear tens of seconds, making it nearly unusable.

**Option 1:**

Disabling addon-a11y "resolves" the issue for us as well.

```ts
const config: StorybookConfig = {
  addons: [
    // '@storybook/addon-a11y',
    ...
  ],
  ...
};

```

**Option 2:**

Setting `globals.a11y.manual` to `true` (as suggested by @smlimasu) fixes the issue.

```ts
const preview: Preview = {
  parameters: {
    globals: {
      a11y: {
        // Optional flag to prevent the automatic check
        // It fixes the issue with the a11y addon freezing
        // cf. https://github.com/storybookjs/storybook/issues/30975
        manual: true,
      },
    },
  }
}
```

However, according to @valentinpalkovic 's comment [here](https://github.com/storybookjs/storybook/issues/30314#issuecomment-2775681553), this setting will be removed in v9. So it's probably not a future-proof solution.
> Not relevant anymore, since `parameters.a11y.manual` does not exist anymore in SB 9.0

**Comment by shilman** [comment]
@guicara Typo in your comment. I think you mean `globals.a11y.manual` not `parameters.globals.a11y.manual`

**Comment by guicara** [comment]
@shilman Indeed, you're right!

**Comment by hernan-yadiel** [comment]
I saw this add-on clearly blocking the main thread on Chrome's Performance profiler. TY!
